tests/src/Kernel/BundlePluginTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\Tests\entity\Kernel;
+
+use Drupal\entity_module_bundle_plugin_test\Entity\EntityTestBundlePlugin;
+use Drupal\KernelTests\KernelTestBase;
+
+/**
+ * Tests the bundle plugin API.
+ *
+ * @group entity
+ */
+class BundlePluginTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'system',
+    'entity',
+    'entity_module_bundle_plugin_test',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->installSchema('system', 'router');
+    $this->installEntitySchema('entity_test_bundle_plugin');
+    $this->installConfig('entity_module_bundle_plugin_test');
+  }
+
+  /**
+   * Tests the bundle plugins.
+   */
+  public function testPluginBundles() {
+    $bundled_entity_types = entity_get_bundle_plugin_entity_types();
+    /** @var \Drupal\Core\Entity\EntityTypeInterface $entity_type */
+    $entity_type = $bundled_entity_types['entity_test_bundle_plugin'];
+    $this->assertEquals('entity_test_bundle_plugin', $entity_type->id());
+    $this->assertTrue($entity_type->hasHandlerClass('bundle_plugin'));
+
+    /** @var \Drupal\Core\Entity\EntityTypeBundleInfo $entity_type_bundle_info */
+    $entity_type_bundle_info = \Drupal::service('entity_type.bundle.info');
+    $bundle_info = $entity_type_bundle_info->getBundleInfo('entity_test_bundle_plugin');
+    $this->assertEquals(2, count($bundle_info));
+    $this->assertTrue(isset($bundle_info['first']));
+    $this->assertTrue(isset($bundle_info['second']));
```

Comment:
could use ->assertArrayHasKey() for better readability and verbosity should the test ever fail.